daemon/commit.go

```diff
@@ -20,7 +22,18 @@ type ContainerCommitConfig struct {
 // The image can optionally be tagged into a repository.
 func (daemon *Daemon) Commit(container *Container, c *ContainerCommitConfig) (*image.Image, error) {
 	if c.Pause && !container.isPaused() {
-		container.pause()
+		if err := container.pause(); err != nil {
+			switch err.(type) {
+			case errcode.Error:
+				daError, _ := err.(errcode.Error)
+				if daError.ErrorCode() != derr.ErrorCodeNotRunning && daError.ErrorCode() != derr.ErrorCodeAlreadyPaused {
```

Comment:
Now, we just unpause it in the defer if the container was really paused.
When committing we want to pause container, because we want the container keep unchanged.
So, I think the container wheter is already stopped or paused satisfies our requirement